lib/peopleHR.ts

```diff
@@ -71,6 +72,81 @@ async function parseXml(xmlString: string): Promise<any> {
     );
 }
 
+/**
+ * Pulls the PeopleHR RSS feed data from the cache,
+ * or fetches it from the feed if the cache has expired
+ * @returns JSON of the RSS feed
+ */
+async function getJobPostingData() {
+    const cacheFilePath = path.join(
+        process.cwd(),
+        'files',
+        'peopleHRcache.json',
+    );
+
+    if (existsSync(cacheFilePath)) {
+        const cacheFile = readFileSync(cacheFilePath, 'utf8');
+        const cacheJSON = JSON.parse(cacheFile);
+        const cacheTTL = 10 * 1000; //One hour 60 * 60 * 1000
+
+        if (cacheJSON.lastUpdated + cacheTTL < Date.now()) {
+            // The cache has timed out, fetch new data from PeopleHR
+
+            const peopleHRJobPostingsXML = await fetchPeopleHRFeed();
+
+            if (peopleHRJobPostingsXML === null) {
+                // In the event of a server error, don't update the job listings
+                // if we have cached listings still available to use.
+
+                const fileData = {
+                    lastUpdated: Date.now(),
+                    jobJSON: cacheJSON.jobJSON,
+                };
+
+                writeFileSync(cacheFilePath, JSON.stringify(fileData));
+
+                return cacheJSON.jobJSON;
+            } else {
+                // If we've received a 200 code from the server, update according to the feed contents
+                const peopleHRJobPostings = await parseXml(
+                    peopleHRJobPostingsXML,
+                );
+
+                const fileData = {
+                    lastUpdated: Date.now(),
+                    jobJSON: peopleHRJobPostings,
+                };
+
+                writeFileSync(cacheFilePath, JSON.stringify(fileData));
+
+                return peopleHRJobPostings;
+            }
+        } else {
+            // The cache has not timed out yet, return the cached JSON
+            return cacheJSON.jobJSON;
+        }
+    } else {
+        // If the cache file doesn't exist, create one
+        const peopleHRJobPostingsXML = await fetchPeopleHRFeed();
```

Comment:
**Suggestion**: Similar to the parseXml method, we'll have to catch errors if we want to resolve them quitely.

Perhaps we can replace `if (peopleHRJobPostingsXML === null) {` with the snippet below
```suggestion
        const peopleHRJobPostingsXML = await fetchPeopleHRFeed().catch(function(e) {
    console.error(e);
    return null;
});
```